pkg/kms/localkms/internal/util/localkms_aead.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package util
+
+import (
+	"encoding/base64"
+
+	"github.com/google/tink/go/tink"
+
+	"github.com/hyperledger/aries-framework-go/pkg/secretlock"
+)
+
+// AEAD represents a local kms aead service invoking a local SecretLock to a particular key URI.
+// Instances of AEAD are invoked internally by Tink for wrapping/unwrapping keys. It must not
+// be used elsewhere.
+type AEAD struct {
+	keyURI     string
+	secretLock secretlock.Service
+}
+
+// newLocalStorageAEAD returns a new localstorage AEAD service containing a local SecretLock service.
+func newLocalStorageAEAD(keyURI string, secLock secretlock.Service) tink.AEAD {
+	return &AEAD{
+		keyURI:     keyURI,
+		secretLock: secLock,
+	}
+}
+
+// Encrypt AEAD encrypts the plaintext data and uses addtionaldata from authentication.
+func (a *AEAD) Encrypt(plaintext, additionalData []byte) ([]byte, error) {
+	req := &secretlock.EncryptRequest{
+		Plaintext:                   base64.URLEncoding.EncodeToString(plaintext),
+		AdditionalAuthenticatedData: base64.URLEncoding.EncodeToString(additionalData),
+	}
+
+	resp, err := a.secretLock.Encrypt(a.keyURI, req)
+	if err != nil {
+		return nil, err
+	}
+
+	ct, err := base64.URLEncoding.DecodeString(resp.Ciphertext)
```

Comment:
If you immediately decode the base64 result of secretlock encrypt, why is it being base64 encoded before being returned? (I'm not suggesting you have to change it, just want to see if it makes sense)

Reply:
the KMS cannot guarantee the secretlock will return an encoded cipher (the framework can accept external implementations of secretlock), so it does its own encoding